pkg/cmd/pulumi/new.go

```diff
@@ -506,13 +506,30 @@ func newNewCmd() *cobra.Command {
 	return cmd
 }
 
+// File or directory names that are considered invisible
+// when considering whether a directory is empty.
+var invisibleDirEntries = map[string]struct{}{
+	".git": {},
+	".hg":  {},
+	".bzr": {},
+}
+
 // errorIfNotEmptyDirectory returns an error if path is not empty.
 func errorIfNotEmptyDirectory(path string) error {
 	infos, err := os.ReadDir(path)
 	if err != nil {
 		return err
 	}
 
+	cleanInfos := infos[:0] // zero-alloc filter
```

Comment:
Does that matter? Doesn't the append below just cause an allocate?

Reply:
No the append below doesn't cause an allocation.
The new slice has length zero but capacity of whatever the backing store of the original slice was.
This basically overwrites over them.
This pattern comes from https://github.com/golang/go/wiki/SliceTricks#filtering-without-allocating.
I've written a [short post](https://abhinavg.net/2019/07/11/zero-alloc-slice-filter/) in the past to help visualize this.

**However**, it looks like my brain wasn't at 100% when I made this change because we don't actually need a filtered slice. We just need a count. I've updated the change.

Reply:
> We just need a count. I've updated the change.

Don't actually even need a count, it's just a flag. Soon as you find one you can break.

Reply:
Indeed.